extensions/console/internal.m

```diff
@@ -0,0 +1,382 @@
+/// Parameters:
+///  * None
+///
+/// Returns:
+///  * an hs.window object
+static int console_asWindow(lua_State *L) {
+    NSWindow *console = [[MJConsoleWindowController singleton] window] ;
+    CGWindowID windowID = (CGWindowID)[console windowNumber];
+    [[LuaSkin shared] requireModule:"hs.window"] ;
+    lua_getfield(L, -1, "windowForID") ;
+    lua_pushinteger(L, windowID) ;
+    lua_call(L, 1, 1) ;
+    return 1 ;
+}
+
+/// hs.console.windowBackgroundColor([color]) -> color
+/// Function
+/// Get or set the color for the background of the Hammerspoon Console's window.
+///
+/// Parameters:
+/// * color - an optional table containing color keys as described in `hs.drawing.color`
+///
+/// Returns:
+/// * the current color setting as a table
+///
+/// Notes:
+///  * See the `hs.drawing.color` entry in the Dash documentation, or type `help.hs.drawing.color` in the Hammerspoon console to get more information on how to specify a color.
+static int console_backgroundColor(lua_State *L) {
+    NSWindow *console = [[MJConsoleWindowController singleton] window] ;
+
+    if (lua_type(L, 1) != LUA_TNONE) {
+        luaL_checktype(L, 1, LUA_TTABLE) ;
+        [console setBackgroundColor:[[LuaSkin shared] luaObjectAtIndex:1 toClass:"NSColor"]] ;
+    }
+
+    [[LuaSkin shared] pushNSObject:[console backgroundColor]] ;
+    return 1 ;
+}
+
+/// hs.console.outputBackgroundColor([color]) -> color
+/// Function
+/// Get or set the color for the background of the Hammerspoon Console's output view.
+///
+/// Parameters:
+/// * color - an optional table containing color keys as described in `hs.drawing.color`
+///
+/// Returns:
+/// * the current color setting as a table
+///
+/// Notes:
+///  * See the `hs.drawing.color` entry in the Dash documentation, or type `help.hs.drawing.color` in the Hammerspoon console to get more information on how to specify a color.
+static int console_outputBackgroundColor(lua_State *L) {
+    NSTextView *output = [MJConsoleWindowController singleton].outputView ;
+
+    if (lua_type(L, 1) != LUA_TNONE) {
+        luaL_checktype(L, 1, LUA_TTABLE) ;
+        [output setBackgroundColor:[[LuaSkin shared] luaObjectAtIndex:1 toClass:"NSColor"]] ;
+    }
+
+    [[LuaSkin shared] pushNSObject:[output backgroundColor]] ;
+    return 1 ;
+}
+
+/// hs.console.inputBackgroundColor([color]) -> color
+/// Function
+/// Get or set the color for the background of the Hammerspoon Console's input field.
+///
+/// Parameters:
+/// * color - an optional table containing color keys as described in `hs.drawing.color`
+///
+/// Returns:
+/// * the current color setting as a table
+///
+/// Notes:
+///  * See the `hs.drawing.color` entry in the Dash documentation, or type `help.hs.drawing.color` in the Hammerspoon console to get more information on how to specify a color.
+static int console_inputBackgroundColor(lua_State *L) {
+    NSTextField *input = [MJConsoleWindowController singleton].inputField ;
+
+    if (lua_type(L, 1) != LUA_TNONE) {
```

Comment:
Is there a reason to not be using `LuaSkin checkArgs` for the various functions here that have arguments?